CMSgov/design-system
# [NO-TICKET] Convert Alert to functional component

## Summary

- Converted Alert to a functional component
- Created new `mergeRefs` and `useAutoFocus` utilities
    - There are other places in our codebase where we can use `mergeRefs` to avoid some messy and verbose code
    - We will need `useAutoFocus` when we're migrating components away from `FormControl`
- Moved analytics logic into a separate `useAlertAnalytics` hook that makes use of the newish `useAnalyticsContent` hook
    - This would simplify the logic, but I added some glue code that makes it match the output of the old logic that gathered analytics content. I wanted to avoid making any behavioral changes with this unticketed refactor, but we should consider adding this to our upcoming breaking-change release
- Added a few more checks to the unit tests

## How to test

1. `yarn storybook`
2. Navigate to http://localhost:6006/?path=/story/components-alert--error-alert&args=autoFocus:false&globals=analytics:on
3. Open your JavaScript console to make sure analytics still fire
4. Try setting the `autoFocus` prop in the story controls. It should focus on the last one in the story (because we have three alerts per story and you can only focus on one)

## Checklist




### If this is a change to code:

- [x] Updated unit test snapshots (`yarn update-snapshots`) and loki reference images **if necessary** (`yarn loki && yarn loki:healthcare && yarn loki:medicare`)
